lib/backend/syncersv1/updateprocessors/bgppeerprocessor.go

```diff
@@ -42,31 +45,45 @@ func convertBGPPeerV2ToV1(kvp *model.KVPair) (*model.KVPair, error) {
 		return nil, errors.New("Value is not a valid BGPPeer resource value")
 	}
 
-	// Correct data types.  Handle the conversion.  Start with the v1 key.  The PeerIP and
-	// the Node are now in the Spec - if a Node is not specified then this is a global
-	// peer.
+	// Correct data types. Handle the conversion.
 	ip := cnet.ParseIP(v3res.Spec.PeerIP)
-	if ip == nil {
-		return nil, errors.New("PeerIP is not assigned or is malformed")
+	if ip == nil && v3res.Spec.PeerSelector == "" {
+		return nil, errors.New("Peer IP or selector is not assigned or is malformed")
```

Comment:
It would be nicer to have a precise error here; one of:
- Either Peer IP or Peer Selector must be specified.
- Peer IP is malformed
- Peer Selector is malformed